src/containers/Bookings.js

```diff
@@ -1,19 +1,62 @@
-import React, { Component } from 'react';
-import Search from '../components/Search.js';
-// import Results from '../components/Results.js';
-// import FakeBookings from '../data/fakeBookings.json';
+import React, { Component } from "react";
+import Search from "../components/Search.js";
+import BookingsMessage from "../components/BookingsMessage.js";
+import Results from "../components/Results.js";
+import fakeBookings from "../data/fakeBookings.json";
+import Counter from "../components/Counter";
 
 export default class Bookings extends Component {
-  search = () => {
-    console.info('TO DO!');
+  state = {
+    count: 0,
+    count2: 0,
+    sum: 0,
+    input: 0,
+    bookings: fakeBookings
   };
-  
+
+  userInput = e => {
+    e.preventDefault();
+    this.setState({
+      input: parseInt(e.target.value, 10)
+    });
+  };
+
+  bookingsById = () => {
+    this.setState({
+      bookings: fakeBookings.filter(booking => {
+        return booking.id === this.state.input;
+      })
+    });
+  };
+
+  increase = () => {
+    this.setState({
+      count: ++this.state.count,
+      sum: ++this.state.sum
+    });
+  };
+
+  increase2 = () => {
+    this.setState({
+      count2: ++this.state.count2,
+      sum: ++this.state.sum
+    });
+  };
+
   render() {
     return (
       <div className="App-content">
         <div className="container">
-          <Search search={this.search} />
-          {/* <Results results={this.state.results} /> */}
+          <BookingsMessage />
+          <Search userInput={this.userInput} bookingsById={this.bookingsById} />
```

Comment:
same for `userInput` and `bookingsById` - name them more descriptive. Something like `onKeywordChange` and `onSearch` (and I think both of them can just be one function onSearch maybe? you can just get the value of the text box when they click the button if you're not filtering the list on the fly)

Reply:
If I join them into one function I am not sure if I can take value from an input after pressing the button. What I am doing here is that I am passing these two function to Search component and in Seach component there are input and button to which I pass each function. I don't remember if I can know the value of input in onClick function of a button.

Reply:
you're right .. forget what I said about getting value. Just name them with something more clear and related to Search